scrubber.go

```diff
@@ -0,0 +1,41 @@
+// Scrubber provides a streaming metadata remover
+package metascrubber
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+
+	"github.com/getlantern/meta-scrubber/jpeg"
+	"github.com/getlantern/meta-scrubber/png"
+)
+
+func GetScrubber(reader io.Reader) (io.Reader, error) {
+	// net/http.sniffLen = 512
+	contentTypeHead := make([]byte, 512)
+	n, err := io.ReadFull(reader, contentTypeHead)
+
+	if err != nil {
+		if err == io.EOF {
+			// TODO: what's the best way to return an empty reader
+			return bytes.NewReader([]byte{}), nil
+		} else if err == io.ErrUnexpectedEOF {
+			return bytes.NewReader(contentTypeHead[:n]), nil
+		} else {
+			return nil, err
+		}
+	}
```

Comment:
Also, it might be helpful to add context to this error, like `fmt.Errorf("unable to read content type: %w", err)`.  Might be good to reset the reader in that case too.  That way the caller can proceed to use the non-scrubbing reader if that's okay for their use case.

Reply:
I was trying to think of what an error attempting to read here would represent. It seems to me that it would be something upstream, like reading from the network or a file or ?? I'm still getting a handle on go idioms, but it seems like I'd want to return the error undiluted??

Reply:
Actually, I find that it's a bit rare that it makes sense to return an error undiluted.  The big difference between Go and other languages here is that it's not necessarily easy to get the whole stack trace of the error.  I think attaching context is a way to achieve a similar outcome as it gives you some idea of the call stack that lead to the error.  Don't get me wrong, it definitely seems worse than just normal error-tracing, but it's what we're stuck with.

The [new Go errors API](https://blog.golang.org/go1.13-errors) should help a bit.  It also further incentivizes error-wrapping since callers can use `errors.As` and `errors.Is` to search for possible causes in the call stack.

Thinking about this specific case a bit more, it may make sense to attach the package name to the error here.  TBH, I don't see that too much outside of the standard library, but it seems conceivable that someone could be given a reader from this package and have little to no knowledge of where it came from.  To them, this just represents JPEG data.  If they get an error like `upload failed: read failed: metascrubber: invalid JPEG structure`, it gives them some idea of where to look.